File: scripts/php/f20.loop.php
#!/usr/bin/env php
<?php
// conditional loop with do..while
do {
    echo "Enter your name (quit to Exit): "; # output prompt
    $answer = rtrim(fgets(STDIN));           # get input, trim newline
    if ($answer != "quit")
        echo "Hello $answer!\n";             # output result if not exiting
} while ($answer != "quit");
?>

File: compiled_lang/go/src/f20.loop.go
// testbox: title="while-style for (single condition)"

package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	stdin := bufio.NewScanner(os.Stdin)
	answer := ""
	for answer != "quit" {
		fmt.Print("Enter your name (quit to Exit): ")
		stdin.Scan()
		answer = stdin.Text()

		if answer != "quit" {
			fmt.Println("Hello " + answer + "!")
		}
	}
}
